packages/dendron-cli/src/commands/publishCLICommand.ts

```diff
@@ -414,14 +408,39 @@ export class PublishCLICommand extends CLICommand<CommandOpts, CommandOutput> {
     });
   }
 
-  async _initialize(opts: { nextPath: string; spinner: ora.Ora }) {
-    const { spinner } = opts;
+  /**
+   * Clone template
+   * If templateVersion is set in dendron.yml, clone at version
+   * Otherwise, get same version as engine
+   * @param opts
+   * @return: string, version of template
+   */
+  async _initialize(opts: {
+    nextPath: string;
+    spinner: ora.Ora;
+    wsRoot: string;
+  }) {
+    const { spinner, wsRoot } = opts;
     SpinnerUtils.renderAndContinue({
       spinner,
       text: "Initializing NextJS template.",
     });
+    const config = DConfig.readConfigAndApplyLocalOverrideSync(opts.wsRoot);
     await this._cloneTemplate(opts);
+    const templateVersion =
+      NextjsExportPodUtils.templateVersion(config) ||
```

Comment:
What happens if config version exceeds engine Version?  Shouldn't we do `Math.min(configVersion?, engineVersion)` ?

Reply:
we don't want to do a min in this case because `configVersion` is source of truth. if the engineVersion is lower, we need to ask the user to update the engine because they've published using a higher version. i've done a writeup of this here, will also share as an async > https://docs.google.com/document/d/1dZ3X039TqQyUMaRYx9KJSFVpJMEHgtjRbqdAdFDGWX8/edit